pkg/scaffold/scaffold.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Modified from github.com/kubernetes-sigs/controller-tools/pkg/scaffold/scaffold.go
+
+package scaffold
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/operator-framework/operator-sdk/pkg/scaffold/input"
+	"github.com/operator-framework/operator-sdk/pkg/util"
+	"golang.org/x/tools/imports"
+)
+
+// Scaffold writes Templates to scaffold new files
+type Scaffold struct {
+	// Repo is the go project package
+	Repo string
+
+	// ProjectPath is the relative path to the project root
+	ProjectPath string
+
+	// ProjectName is the operator's name, ex. app-operator
+	ProjectName string
+
+	GetWriter func(path string, mode os.FileMode) (io.Writer, error)
+}
+
+func (s *Scaffold) setFieldsAndValidate(t input.File) error {
+	if b, ok := t.(input.Repo); ok {
+		b.SetRepo(s.Repo)
+	}
+	if b, ok := t.(input.ProjectPath); ok {
+		b.SetProjectPath(s.ProjectPath)
+	}
+	if b, ok := t.(input.ProjectName); ok {
+		b.SetProjectName(s.ProjectName)
+	}
+
+	// Validate the template is ok
+	if v, ok := t.(input.Validate); ok {
+		if err := v.Validate(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *Scaffold) configure(cfg *input.Config) error {
+	s.Repo = cfg.Repo
+	s.ProjectPath = cfg.ProjectPath
+	s.ProjectName = cfg.ProjectName
+	return nil
```

Comment:
No need to return error if there is none.

Reply:
There might be in the future, if we add more thorough configuration checks. I'll remove for now until error return is required.